From 5d6e09428e2268d82d144641364a69abb1008899 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Thu, 12 Feb 2015 08:59:00 +0530
Subject: [PATCH] pmic_ccsm: On boot, enable internal detection

On boot, keep external phy in reset to enable internal detection

Change-Id: I457474d9ca9501ac17af3b4efca810f4eda9977b
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index c654764..9a2844c 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1631,6 +1631,9 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 		if (!ret && (val & USBPATH_USBSEL_MASK)) {
 				dev_info(chc.dev, "SOC-Internal-USBPHY used\n");
 				chc.is_internal_usb_phy = true;
+				/* Enable internal detection */
+				pmic_write_reg(
+					chc.reg_map->pmic_usbphyctrl, 0x0);
 		} else {
 				dev_info(chc.dev, "External-USBPHY used\n");
 		}
-- 
1.9.1

